--- a/net/minecraft/client/renderer/GameRenderer.java
+++ b/net/minecraft/client/renderer/GameRenderer.java
@@ -1878,8 +1878,6 @@
    }
 
    public static record ResourceCache(ResourceProvider original, Map<ResourceLocation, Resource> cache) implements ResourceProvider {
-      private final ResourceProvider original;
-      private final Map<ResourceLocation, Resource> cache;
 
       public Optional<Resource> getResource(ResourceLocation p_251007_) {
          Resource resource = this.cache.get(p_251007_);
