--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -571,7 +571,7 @@
 
             return flag;
          });
-      }).thenApplyAsync((p_288112_2_) -> {
+      }).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync((p_288112_2_) -> {
          this.level.getProfiler().incrementCounter("chunkLoad");
          if (p_288112_2_.isPresent()) {
             ChunkAccess chunkaccess = ChunkSerializer.read(this.level, this.poiManager, pChunkPos, p_288112_2_.get());
