--- a/net/optifine/model/ModelUtils.java
+++ b/net/optifine/model/ModelUtils.java
@@ -33,11 +33,10 @@
       }
    }
 
-   private static void dbgQuads(String name, List quads, String prefix) {
+   private static void dbgQuads(String name, List<BakedQuad> quads, String prefix) {
       for(BakedQuad bakedquad : quads) {
          dbgQuad(name, bakedquad, prefix);
       }
-
    }
 
    public static void dbgQuad(String name, BakedQuad quad, String prefix) {
@@ -83,14 +82,12 @@
    }
 
    public static List duplicateQuadList(List<BakedQuad> list) {
-      List<BakedQuad> list = new ArrayList<>();
-
-      for(BakedQuad bakedquad : list) {
-         BakedQuad bakedquad1 = duplicateQuad(bakedquad);
-         list.add(bakedquad1);
+      ArrayList<BakedQuad> list2 = new ArrayList<BakedQuad>();
+      for (BakedQuad quad : list) {
+         BakedQuad quad2 = ModelUtils.duplicateQuad(quad);
+         list2.add(quad2);
       }
-
-      return list;
+      return list2;
    }
 
    public static BakedQuad duplicateQuad(BakedQuad quad) {
