--- a/net/minecraft/client/renderer/blockentity/BlockEntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/blockentity/BlockEntityRenderDispatcher.java
@@ -51,7 +51,7 @@
 
    @Nullable
    public <E extends BlockEntity> BlockEntityRenderer<E> getRenderer(E pBlockEntity) {
-      return this.renderers.get(pBlockEntity.getType());
+      return (BlockEntityRenderer<E>)this.renderers.get(pBlockEntity.getType());
    }
 
    public void prepare(Level pLevel, Camera pCamera, HitResult pCameraHitResult) {
@@ -151,7 +151,7 @@
       return this.context;
    }
 
-   public Map<BlockEntityType, BlockEntityRenderer> getBlockEntityRenderMap() {
+   public Map getBlockEntityRenderMap() {
       if (this.renderers instanceof ImmutableMap) {
          this.renderers = new HashMap<>(this.renderers);
       }
