--- a/net/minecraft/client/Options.java
+++ b/net/minecraft/client/Options.java
@@ -2152,7 +2152,7 @@
          int k = indexOf(this.ofTelemetry, OF_TELEMETRY);
          return s + getTranslation(KEYS_TELEMETRY, k);
       } else if (par1EnumOptions.isProgressOption()) {
-         double d0 = par1EnumOptions.get();
+         double d0 = (double) par1EnumOptions.get();
          return d0 == 0.0D ? s + I18n.get("options.off") : s + (int)(d0 * 100.0D) + "%";
       } else {
          return null;
