--- a/net/optifine/shaders/Shaders.java
+++ b/net/optifine/shaders/Shaders.java
@@ -1091,7 +1091,7 @@
       Set set = props.keySet();
       List<ICustomTexture> list = new ArrayList<>();
 
-      for(String s1 : set) {
+      for(String s1 : (Set<String>)(Set<?>)set) {
          if (s1.startsWith(s)) {
             String s2 = StrUtils.removePrefix(s1, s);
             s2 = StrUtils.removeSuffix(s2, new String[]{".0", ".1", ".2", ".3", ".4", ".5", ".6", ".7", ".8", ".9"});
@@ -5113,7 +5113,7 @@
                   Lang.loadLocaleData(inputstream, properties);
                   inputstream.close();
 
-                  for(String s5 : properties.keySet()) {
+                  for(String s5 : (Set<String>)(Set<?>)properties.keySet()) {
                      String s6 = properties.getProperty(s5);
                      shaderPackResources.put(s5, s6);
                   }
