--- a/net/minecraft/client/particle/ParticleEngine$1ParticleDefinition.java
+++ b/net/minecraft/client/particle/ParticleEngine$1ParticleDefinition.java
@@ -5,8 +5,6 @@
 import net.minecraft.resources.ResourceLocation;
 
 record ParticleEngine$1ParticleDefinition(ResourceLocation id, Optional<List<ResourceLocation>> sprites) {
-   private final ResourceLocation id;
-   private final Optional<List<ResourceLocation>> sprites;
 
    public ResourceLocation id() {
       return this.id;
