--- a/net/minecraft/client/model/geom/ModelPart.java
+++ b/net/minecraft/client/model/geom/ModelPart.java
@@ -351,7 +351,7 @@
       }
    }
 
-   public ModelPart getChild(String id) {
+   public ModelPart getChildByID(String id) {
       if (id == null) {
          return null;
       } else {
@@ -372,7 +372,7 @@
       if (id == null) {
          return null;
       } else {
-         ModelPart modelpart = this.getChild(id);
+         ModelPart modelpart = this.getChildByID(id);
          if (modelpart != null) {
             return modelpart;
          } else {
