--- a/net/optifine/CustomItems.java
+++ b/net/optifine/CustomItems.java
@@ -424,9 +424,9 @@
       CustomItemProperties[][] acustomitemproperties = new CustomItemProperties[list.size()][];
 
       for(int i = 0; i < list.size(); ++i) {
-         List list = (List)list.get(i);
-         if (list != null) {
-            CustomItemProperties[] acustomitemproperties1 = list.toArray(new CustomItemProperties[list.size()]);
+         List lista = (List)list.get(i);
+         if (lista != null) {
+            CustomItemProperties[] acustomitemproperties1 = (CustomItemProperties[]) lista.toArray(new CustomItemProperties[lista.size()]);
             Arrays.sort(acustomitemproperties1, new CustomItemsComparator());
             acustomitemproperties[i] = acustomitemproperties1;
          }
@@ -482,13 +482,13 @@
          list.add((List<CustomItemProperties>)null);
       }
 
-      List<CustomItemProperties> list = list.get(id);
-      if (list == null) {
-         list = new ArrayList<>();
-         list.set(id, list);
+      List<CustomItemProperties> lista = list.get(id);
+      if (lista == null) {
+         lista = new ArrayList<>();
+         list.set(id, lista);
       }
 
-      list.add(cp);
+      lista.add(cp);
    }
 
    public static BakedModel getCustomItemModel(ItemStack itemStack, BakedModel model, ResourceLocation modelLocation, boolean fullModel) {
