--- a/net/minecraft/client/gui/font/providers/BitmapProvider.java
+++ b/net/minecraft/client/gui/font/providers/BitmapProvider.java
@@ -50,10 +50,6 @@
    }
 
    public static record Definition(ResourceLocation file, int height, int ascent, int[][] codepointGrid) implements GlyphProviderDefinition {
-      private final ResourceLocation file;
-      private final int height;
-      private final int ascent;
-      private final int[][] codepointGrid;
       private static final Codec<int[][]> CODEPOINT_GRID_CODEC = ExtraCodecs.validate(Codec.STRING.listOf().xmap((listIn) -> {
          int i = listIn.size();
          int[][] aint = new int[i][];
@@ -190,15 +186,9 @@
       }
    }
 
-   static record Glyph(float scale, NativeImage texture, int unpackSkipPixels, int unpackSkipRows, int width, int height, int advanceWidth, int ascent) implements GlyphInfo {
-      private final float scale;
-      private final NativeImage image;
-      private final int offsetX;
-      private final int offsetY;
-      private final int width;
-      private final int height;
-      private final int advance;
-      private final int ascent;
+   //static record Glyph(float scale, NativeImage texture, int unpackSkipPixels, int unpackSkipRows, int width, int height, int advanceWidth, int ascent) implements GlyphInfo {
+
+   static record Glyph(float scale, NativeImage image, int offsetX, int offsetY, int width, int height, int advance, int ascent) implements GlyphInfo {
 
       public float getAdvance() {
          return (float)this.advance;
