--- a/net/optifine/shaders/BlockAliases.java
+++ b/net/optifine/shaders/BlockAliases.java
@@ -152,7 +152,7 @@
             Config.dbg("[Shaders] Parsing block mappings: " + path);
             ConnectedParser connectedparser = new ConnectedParser("Shaders");
 
-            for(String s : properties.keySet()) {
+            for(String s : (Set<String>)(Set<?>) properties.keySet()) {
                String s1 = properties.getProperty(s);
                if (s.startsWith("layer.")) {
                   if (blockLayerPropertes == null) {
@@ -255,8 +255,8 @@
                         String s2 = astring1[1];
                         String s3 = astring1[2];
                         String s4 = "{Name:'" + s3 + "'";
-                        List<String> list2 = list1.stream().filter((s) -> {
-                           return s.startsWith(s4);
+                        List<String> list2 = list1.stream().filter((ss) -> {
+                           return ss.startsWith(s4);
                         }).collect(Collectors.toList());
                         if (list2.size() <= 0) {
                            Config.warn("Block not processed: " + s1);
