--- a/net/minecraft/client/renderer/texture/Stitcher.java
+++ b/net/minecraft/client/renderer/texture/Stitcher.java
@@ -125,9 +125,6 @@
    }
 
    static record Holder<T extends Stitcher.Entry>(T entry, int width, int height) {
-      private final T entry;
-      private final int width;
-      private final int height;
 
       public Holder(T pEntry, int pMipLevel) {
          this(pEntry, Stitcher.smallestFittingMinTexel(pEntry.width(), pMipLevel), Stitcher.smallestFittingMinTexel(pEntry.height(), pMipLevel));
