--- a/net/minecraft/world/level/entity/EntitySection.java
+++ b/net/minecraft/world/level/entity/EntitySection.java
@@ -45,7 +45,7 @@
       } else {
          for(T t : collection) {
             U u = (U)((EntityAccess)pTest.tryCast(t));
-            if (u != null && t.getBoundingBox().intersects(pBounds) && pConsumer.accept((T)u).shouldAbort()) {
+            if (u != null && t.getBoundingBox().intersects(pBounds) && pConsumer.accept(u).shouldAbort()) {
                return AbortableIterationConsumer.Continuation.ABORT;
             }
          }
