--- a/net/optifine/http/HttpPipeline.java
+++ b/net/optifine/http/HttpPipeline.java
@@ -162,7 +162,7 @@
    }
 
    public static boolean hasActiveRequests() {
-      for(HttpPipelineConnection httppipelineconnection : mapConnections.values()) {
+      for(HttpPipelineConnection httppipelineconnection : (Iterable<HttpPipelineConnection>) mapConnections.values()) {
          if (httppipelineconnection.hasActiveRequests()) {
             return true;
          }
