--- a/net/optifine/ConnectedProperties.java
+++ b/net/optifine/ConnectedProperties.java
@@ -1,11 +1,7 @@
 package net.optifine;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
+import java.util.*;
+
 import net.minecraft.client.renderer.RenderType;
 import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
 import net.minecraft.client.renderer.texture.TextureAtlas;
@@ -254,7 +250,7 @@
             list.add(s);
          }
 
-         String[] astring2 = list.toArray(new String[list.size()]);
+         String[] astring2 = (String[]) list.toArray(new String[list.size()]);
 
          for(int i1 = 0; i1 < astring2.length; ++i1) {
             String s1 = astring2[i1];
@@ -358,7 +354,7 @@
    }
 
    public static Property getProperty(String key, Collection properties) {
-      for(Property property : properties) {
+      for(Property property : (Collection<Property>) properties) {
          if (key.equals(property.getName())) {
             return property;
          }
@@ -875,7 +871,7 @@
             }
          }
 
-         return list.toArray(new TextureAtlasSprite[list.size()]);
+         return (TextureAtlasSprite[]) list.toArray(new TextureAtlasSprite[list.size()]);
       }
    }
 
