--- a/net/optifine/CustomGuiProperties.java
+++ b/net/optifine/CustomGuiProperties.java
@@ -3,6 +3,8 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Properties;
+import java.util.Set;
+
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.chat.Component;
@@ -174,7 +176,7 @@
 
       String s5 = property + ".";
 
-      for(String s1 : props.keySet()) {
+      for(String s1 : (Set<String>)(Set<?>) props.keySet()) {
          if (s1.startsWith(s5)) {
             String s2 = s1.substring(s5.length());
             s2 = s2.replace('\\', '/');
