--- a/net/optifine/CustomItemProperties.java
+++ b/net/optifine/CustomItemProperties.java
@@ -202,7 +202,7 @@
             }
          }
 
-         Integer[] ainteger = set.toArray(new Integer[set.size()]);
+         Integer[] ainteger = (Integer[]) set.toArray(new Integer[set.size()]);
          int[] aint = new int[ainteger.length];
 
          for(int k = 0; k < aint.length; ++k) {
@@ -272,7 +272,7 @@
          Set set = map.keySet();
          Map map1 = new LinkedHashMap();
 
-         for(String s1 : set) {
+         for(String s1 : (Set<String>)(Set<?>) set) {
             String s2 = (String)map.get(s1);
             s2 = fixTextureName(s2, basePath);
             map1.put(s1, s2);
@@ -330,7 +330,7 @@
          Set set = map.keySet();
          Map map1 = new LinkedHashMap();
 
-         for(String s1 : set) {
+         for(String s1 : (Set<String>)(Set<?>) set) {
             String s2 = (String)map.get(s1);
             s2 = fixModelName(s2, basePath);
             map1.put(s1, s2);
@@ -484,14 +484,14 @@
       } else {
          List list = new ArrayList();
 
-         for(String s1 : map.keySet()) {
+         for(String s1 : (Set<String>)(Set<?>) map.keySet()) {
             String s2 = (String)map.get(s1);
             String s3 = s1.substring(s.length());
             NbtTagValue nbttagvalue = new NbtTagValue(s3, s2);
             list.add(nbttagvalue);
          }
 
-         NbtTagValue[] anbttagvalue = list.toArray(new NbtTagValue[list.size()]);
+         NbtTagValue[] anbttagvalue = (NbtTagValue[]) list.toArray(new NbtTagValue[list.size()]);
          return anbttagvalue;
       }
    }
@@ -499,7 +499,7 @@
    private static Map getMatchingProperties(Properties props, String keyPrefix) {
       Map map = new LinkedHashMap();
 
-      for(String s : props.keySet()) {
+      for(String s : (Set<String>)(Set<?>) props.keySet()) {
          String s1 = props.getProperty(s);
          if (s.startsWith(keyPrefix)) {
             map.put(s, s1);
@@ -614,7 +614,7 @@
       }
 
       if (this.mapSprites != null) {
-         for(String s : this.mapSprites.keySet()) {
+         for(String s : (Set<String>)(Set<?>) this.mapSprites.keySet()) {
             TextureAtlasSprite textureatlassprite = (TextureAtlasSprite)this.mapSprites.get(s);
             if (textureatlassprite != null) {
                ResourceLocation resourcelocation = textureatlassprite.getName();
