--- a/net/optifine/SmartLeaves.java
+++ b/net/optifine/SmartLeaves.java
@@ -2,6 +2,8 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
+
 import net.minecraft.client.renderer.block.model.BakedQuad;
 import net.minecraft.client.resources.model.BakedModel;
 import net.minecraft.client.resources.model.ModelManager;
@@ -118,7 +120,7 @@
                   } else if (list.size() != 6) {
                      return null;
                   } else {
-                     for(BakedQuad bakedquad : list) {
+                     for(BakedQuad bakedquad : (List<BakedQuad>)list) {
                         List list1 = bakedmodel.getQuads((BlockState)null, bakedquad.getDirection(), RANDOM);
                         if (list1.size() > 0) {
                            return null;
