--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -178,8 +178,6 @@
    }
 
    static record GlyphInfoFilter(GlyphInfo glyphInfo, GlyphInfo glyphInfoNotFishy) {
-      private final GlyphInfo glyphInfo;
-      private final GlyphInfo glyphInfoNotFishy;
       static final FontSet.GlyphInfoFilter MISSING = new FontSet.GlyphInfoFilter(SpecialGlyphs.MISSING, SpecialGlyphs.MISSING);
 
       GlyphInfo select(boolean pFilterFishyGlyphs) {
