--- a/net/minecraft/client/GuiMessage.java
+++ b/net/minecraft/client/GuiMessage.java
@@ -4,11 +4,7 @@
 import net.minecraft.network.chat.MessageSignature;
 import net.minecraft.util.FormattedCharSequence;
 
-public record GuiMessage(int updateCounterCreated, Component content, MessageSignature headerSignature, GuiMessageTag tag) {
-   private final int addedTime;
-   private final Component content;
-   private final MessageSignature signature;
-   private final GuiMessageTag tag;
+public record GuiMessage(int addedTime, Component content, MessageSignature signature, GuiMessageTag tag) {
 
    public int addedTime() {
       return this.addedTime;
@@ -27,10 +23,6 @@
    }
 
    public static record Line(int addedTime, FormattedCharSequence content, GuiMessageTag tag, boolean endOfEntry) {
-      private final int addedTime;
-      private final FormattedCharSequence content;
-      private final GuiMessageTag tag;
-      private final boolean endOfEntry;
 
       public int addedTime() {
          return this.addedTime;
