--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -518,23 +518,20 @@
       path = StrUtils.removeSuffix(path, ".json");
       return StrUtils.removeSuffix(path, ".png");
    }
-
-   public BlockModel loadBlockModel(ResourceLocation locJson) {
-      try {
-         Resource resource = Config.getResource(locJson);
-         Reader reader = resource.openAsReader();
-         return BlockModel.fromStream(reader);
-      } catch (Exception exception) {
-         Config.warn("Error loading model: " + locJson);
-         Config.warn(exception.getClass().getName() + ": " + exception.getMessage());
-         return null;
-      }
-   }
+   //TODO: check it
+//   public BlockModel loadBlockModel(ResourceLocation locJson) {
+//      try {
+//         Resource resource = Config.getResource(locJson);
+//         Reader reader = resource.openAsReader();
+//         return BlockModel.fromStream(reader);
+//      } catch (Exception exception) {
+//         Config.warn("Error loading model: " + locJson);
+//         Config.warn(exception.getClass().getName() + ": " + exception.getMessage());
+//         return null;
+//      }
+//   }
 
    static record BakedCacheKey(ResourceLocation id, Transformation transformation, boolean isUvLocked) {
-      private final ResourceLocation id;
-      private final Transformation transformation;
-      private final boolean isUvLocked;
 
       public ResourceLocation id() {
          return this.id;
@@ -556,8 +553,6 @@
    }
 
    public static record LoadedJson(String source, JsonElement data) {
-      private final String source;
-      private final JsonElement data;
 
       public String source() {
          return this.source;
