--- a/net/optifine/util/ResUtils.java
+++ b/net/optifine/util/ResUtils.java
@@ -96,7 +96,7 @@
             }
          }
 
-         return list.toArray(new String[list.size()]);
+         return (String[]) list.toArray(new String[list.size()]);
       }
    }
 
@@ -132,7 +132,7 @@
             }
          }
 
-         return list.toArray(new String[list.size()]);
+         return (String[]) list.toArray(new String[list.size()]);
       }
    }
 
@@ -160,7 +160,7 @@
          }
 
          zipfile.close();
-         String[] astring = list.toArray(new String[list.size()]);
+         String[] astring = (String[]) list.toArray(new String[list.size()]);
          return astring;
       } catch (IOException ioexception) {
          ioexception.printStackTrace();
