--- a/net/minecraft/world/level/biome/Biome.java
+++ b/net/minecraft/world/level/biome/Biome.java
@@ -314,7 +314,7 @@
       }
    }
 
-   private static record ClimateSettings(boolean hasPrecipitation, float temperature, Biome.TemperatureModifier temperatureModifier, float downfall) {
+   public static record ClimateSettings(boolean hasPrecipitation, float temperature, Biome.TemperatureModifier temperatureModifier, float downfall) {
       public static final MapCodec<Biome.ClimateSettings> CODEC = RecordCodecBuilder.mapCodec((p_264995_) -> {
          return p_264995_.group(Codec.BOOL.fieldOf("has_precipitation").forGetter((p_264996_) -> {
             return p_264996_.hasPrecipitation;
