--- a/net/optifine/util/FontUtils.java
+++ b/net/optifine/util/FontUtils.java
@@ -6,6 +6,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Properties;
+import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 import net.minecraft.Util;
@@ -49,7 +50,7 @@
    public static Int2ObjectMap<Float> readCustomCharWidths(Properties props) {
       Int2ObjectMap<Float> int2objectmap = new Int2ObjectOpenHashMap<>();
 
-      for(String s : props.keySet()) {
+      for(String s : (Set<String>)(Set<?>) props.keySet()) {
          String s1 = "width.";
          if (s.startsWith(s1)) {
             String s2 = s.substring(s1.length());
