--- a/net/minecraft/util/ClassInstanceMultiMap.java
+++ b/net/minecraft/util/ClassInstanceMultiMap.java
@@ -58,7 +58,7 @@
          List<? extends T> list = this.byClass.computeIfAbsent(pType, (p_13537_1_) -> {
             return this.allInstances.stream().filter(p_13537_1_::isInstance).collect(Collectors.toList());
          });
-         return Collections.unmodifiableCollection(list);
+         return (Collection<S>) Collections.unmodifiableCollection(list);
       }
    }
 
