--- a/net/minecraft/client/renderer/texture/SpriteLoader.java
+++ b/net/minecraft/client/renderer/texture/SpriteLoader.java
@@ -198,13 +198,6 @@
    }
 
    public static record Preparations(int width, int height, int mipLevel, TextureAtlasSprite missing, Map<ResourceLocation, TextureAtlasSprite> regions, CompletableFuture<Void> readyForUpload) {
-      private final int width;
-      private final int height;
-      private final int mipLevel;
-      private final TextureAtlasSprite missing;
-      private final Map<ResourceLocation, TextureAtlasSprite> regions;
-      private final CompletableFuture<Void> readyForUpload;
-
       public CompletableFuture<SpriteLoader.Preparations> waitForUpload() {
          return this.readyForUpload.thenApply((voidIn) -> {
             return this;
