--- a/net/optifine/entity/model/CustomEntityModels.java
+++ b/net/optifine/entity/model/CustomEntityModels.java
@@ -207,7 +207,7 @@
 
    private static Map<EntityType, EntityRenderer> getEntityRenderMap() {
       EntityRenderDispatcher entityrenderdispatcher = Minecraft.getInstance().getEntityRenderDispatcher();
-      Map<EntityType, EntityRenderer> map = entityrenderdispatcher.getEntityRenderMap();
+      Map map = entityrenderdispatcher.getEntityRenderMap();
       if (map == null) {
          return null;
       } else {
