--- a/net/minecraft/client/particle/ParticleEngine.java
+++ b/net/minecraft/client/particle/ParticleEngine.java
@@ -326,7 +326,7 @@
 
    @Nullable
    private <T extends ParticleOptions> Particle makeParticle(T pParticleData, double pX, double pY, double pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
-      ParticleProvider<T> particleprovider = this.providers.get(BuiltInRegistries.PARTICLE_TYPE.getKey(pParticleData.getType()));
+      ParticleProvider<T> particleprovider = (ParticleProvider<T>) this.providers.get(BuiltInRegistries.PARTICLE_TYPE.getKey(pParticleData.getType()));
       return particleprovider == null ? null : particleprovider.createParticle(pParticleData, this.level, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
    }
 
@@ -604,9 +604,12 @@
       this.trackedParticleCounts.clear();
    }
 
-   private boolean reuseBarrierParticle(Particle var1, Queue<Particle> var2) {
-      // $FF: Couldn't be decompiled
+   private boolean reuseBarrierParticle(Particle entityfx, Queue<Particle> deque) {
+      for (Particle particle : deque) {
+      }
+      return false;
    }
+
 
    public static void updateTerrainParticleColor(Particle particle, BlockState state, BlockAndTintGetter world, BlockPos pos, RenderEnv renderEnv) {
       renderEnv.reset(state, pos);
