--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -499,8 +499,6 @@
    }
 
    static record DelayedMessageDeletion(MessageSignature signature, int deletableAfter) {
-      private final MessageSignature signature;
-      private final int deletableAfter;
 
       public MessageSignature signature() {
          return this.signature;
