--- a/net/optifine/ConnectedTextures.java
+++ b/net/optifine/ConnectedTextures.java
@@ -1633,7 +1633,7 @@
          }
       }
 
-      ConnectedProperties[] aconnectedproperties1 = list.toArray(new ConnectedProperties[list.size()]);
+      ConnectedProperties[] aconnectedproperties1 = (ConnectedProperties[]) list.toArray(new ConnectedProperties[list.size()]);
       Set set1 = new HashSet();
       Set set = new HashSet();
 
@@ -1653,17 +1653,14 @@
    }
 
    private static ConnectedProperties[][] propertyListToArray(List list) {
-      ConnectedProperties[][] aconnectedproperties = new ConnectedProperties[list.size()][];
-
-      for(int i = 0; i < list.size(); ++i) {
-         List list = (List)list.get(i);
-         if (list != null) {
-            ConnectedProperties[] aconnectedproperties1 = list.toArray(new ConnectedProperties[list.size()]);
-            aconnectedproperties[i] = aconnectedproperties1;
-         }
+      ConnectedProperties[][] propArr = new ConnectedProperties[list.size()][];
+      for (int i = 0; i < list.size(); ++i) {
+         List subList = (List)list.get(i);
+         if (subList == null) continue;
+         ConnectedProperties[] subArr = (ConnectedProperties[]) subList.toArray(new ConnectedProperties[subList.size()]);
+         propArr[i] = subArr;
       }
-
-      return aconnectedproperties;
+      return propArr;
    }
 
    private static void addToTileList(ConnectedProperties cp, List tileList) {
@@ -1700,17 +1697,15 @@
    }
 
    private static void addToList(ConnectedProperties cp, List list, int id) {
-      while(id >= list.size()) {
-         list.add((Object)null);
-      }
-
-      List list = (List)list.get(id);
-      if (list == null) {
-         list = new ArrayList();
-         list.set(id, list);
-      }
-
-      list.add(cp);
+      while (id >= list.size()) {
+         list.add(null);
+      }
+      ArrayList<ConnectedProperties> subList = (ArrayList<ConnectedProperties>)list.get(id);
+      if (subList == null) {
+         subList = new ArrayList<ConnectedProperties>();
+         list.set(id, subList);
+      }
+      subList.add(cp);
    }
 
    private static String[] getDefaultCtmPaths() {
@@ -1730,7 +1725,7 @@
          addDefaultLocation(list, "textures/block/" + s + "_stained_glass.png", s1 + "_glass_" + s + "/glass_pane_" + s + ".properties");
       }
 
-      return list.toArray(new String[list.size()]);
+      return (String[]) list.toArray(new String[list.size()]);
    }
 
    private static void addDefaultLocation(List list, String locBase, String pathSuffix) {
