--- a/net/optifine/NaturalTextures.java
+++ b/net/optifine/NaturalTextures.java
@@ -79,7 +79,7 @@
                }
             }
 
-            propertiesByIndex = arraylist.toArray(new NaturalProperties[arraylist.size()]);
+            propertiesByIndex = (NaturalProperties[]) arraylist.toArray(new NaturalProperties[arraylist.size()]);
             if (i > 0) {
                Config.dbg("NaturalTextures: " + i);
             }
