--- a/net/optifine/TextureAnimations.java
+++ b/net/optifine/TextureAnimations.java
@@ -75,7 +75,7 @@
          }
       }
 
-      return list.toArray(new TextureAnimation[list.size()]);
+      return (TextureAnimation[]) list.toArray(new TextureAnimation[list.size()]);
    }
 
    private static TextureAnimation[] getTextureAnimations(PackResources rp) {
@@ -111,7 +111,7 @@
             }
          }
 
-         return list.toArray(new TextureAnimation[list.size()]);
+         return (TextureAnimation[]) list.toArray(new TextureAnimation[list.size()]);
       }
    }
 
